### DISCUSSION TITLE
Understanding of Wifi Background Tasks

### DISCUSSION THREAD

**Discussion by mirisbowring** [immich-app/immich#3104] [Q&A]
I was waiting for the wifi only backup option on iOS until i discovered the 1.48.0 Release Notes.

Did not know to configure it in the iOS Settings.

But here comes my Question. It seems like if i want immich to only upload in background via Wifi, i have to globally switch it for all apps. Meaning that e.g. Amazon would not sync in background anymore if i am on my way with mobile data only.

Do i have some misunderstanding?

**Comment by alextran1502** [comment]
Correct, this is the limitation at the moment for background tasks

**Reply by alextran1502** [reply]
@martyfuhry You might understand the complication of checking WIFI and syncing status for background task to help explain the situation.

**Reply by martyfuhry** [reply]
The best it seems we can do is check [isExpensive](https://developer.apple.com/documentation/network/nwpath/2998722-isexpensive), as the [connectivity_plus](https://github.com/fluttercommunity/plus_plugins/issues/612) discussion here seems to suggest.

We would put the check [here when we check for the charging requirement](https://github.com/immich-app/immich/blob/main/mobile/ios/Runner/BackgroundSync/BackgroundServicePlugin.swift#L286).

**Comment by pacman789** [comment]
> Correct, this is the limitation at the moment for background tasks

Hi [alextran1502](https://github.com/alextran1502),

Is there any plan to change it? It would be nice  to setup wifi background sync just like in Google Photos.

Thanks for your answer.
